### PR TITLE
Pathfinding cleanup

### PR DESCRIPTION
**Pull request by richard-fine**
Consolidate all the pathfinding code into GuestPathfinding.cpp, and introduce a GuestPathfinding.h for the externally visible members (instead of leaving their declarations in Peep.h). I've also commented what those externally-visible members are / do, to the best of my knowledge.

### CODE REVIEW

**Comment by duncanspumpkin** [comment]
Don't change logic and refactor in one pr. Keep them separate.

**Comment by richard-fine** [comment]
Split the logic changes off into https://github.com/OpenRCT2/OpenRCT2/pull/12917

File: src/openrct2/peep/GuestPathfinding.cpp [hunk 1]
```diff
@@ -2217,3 +2230,81 @@ int32_t guest_path_finding(Guest* peep)
 #endif // defined(DEBUG_LEVEL_1) && DEBUG_LEVEL_1
     return peep_move_one_tile(direction, peep);
 }
+
+bool is_valid_path_z_and_direction(TileElement* tileElement, int32_t currentZ, int32_t currentDirection)
```
Comment (duncanspumpkin) [inline src/openrct2/peep/GuestPathfinding.cpp:2234]: Since you are moving these functions can you go through and follow the coding style. 
Function names should be TitleCase. variable names should be lower camelCase (which I'm pretty sure they all are).

File: src/openrct2/peep/GuestPathfinding.cpp [hunk 2]
```diff
@@ -2217,3 +2230,81 @@ int32_t guest_path_finding(Guest* peep)
...
+            return false;
+    }
+    return true;
+}
+
+/**
+ *
+ *  rct2: 0x0069A98C
+ */
+void peep_reset_pathfind_goal(Peep* peep)
```
Comment (duncanspumpkin) [inline src/openrct2/peep/GuestPathfinding.cpp:2265]: This is not a `Guest` function but a `Peep` function I've never been too sure though what I was doing with splitting things up into a GuestPathfinding as its pretty much all Peep pathfinding. Eitherway this is a member function and should be changed to reflect that.

File: src/openrct2/peep/GuestPathfinding.h [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+ * Copyright (c) 2014-2020 OpenRCT2 developers
+ *
+ * For a complete list of all authors, please refer to contributors.md
+ * Interested in contributing? Visit https://github.com/OpenRCT2/OpenRCT2
+ *
+ * OpenRCT2 is licensed under the GNU General Public License version 3.
+ *****************************************************************************/
+
+#ifndef _GUESTPATHFINDING_H_
+#define _GUESTPATHFINDING_H_
```
Comment (duncanspumpkin) [inline src/openrct2/peep/GuestPathfinding.h:11]: we use #pragma once instead of an include guard

File: src/openrct2/peep/GuestPathfinding.h [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+#include "../common.h"
+#include "../ride/RideTypes.h"
```
Comment (duncanspumpkin) [inline src/openrct2/peep/GuestPathfinding.h:14]: why is this included?

Comment (richard-fine) [reply src/openrct2/peep/GuestPathfinding.h:14]: For `ride_id_t` on line 31.

**Review by duncanspumpkin — APPROVED** [review]
